### ISSUE TITLE
DMI_RANDOM_USED_ONLY_ONCE false negative when using `@NoArgsConstructor`

### ISSUE THREAD

**Issue by Emilyaxe** [spotbugs/spotbugs#4035]
**Describe the bug**
`DMI_RANDOM_USED_ONLY_ONCE` is reported for `new Random().nextInt()` and `nextInt(bound)` when a class has an explicit private constructor, but the warning disappears when the constructor is generated via Lombok’s `@NoArgsConstructor(access = AccessLevel.PRIVATE)`, even though the compiled bytecode is identical. This leads to a false negative.

**To Reproduce**

```java
// BEFORE — DMI_RANDOM_USED_ONLY_ONCE reported twice (correct)
package demo.before;

import java.util.Random;

public class Utility {
    private Utility() {}

    public static int numberProvider() {
        return new Random().nextInt();
    }

    public static int boundedProvider(int bound) {
        return new Random().nextInt(bound);
    }
}
```

```java
// AFTER — no findings (false negative)
package demo.after;

import java.util.Random;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class Utility {

    public static int numberProvider() {
        return new Random().nextInt();
    }

    public static int boundedProvider(int bound) {
        return new Random().nextInt(bound);
    }
}
```

Compile both with Lombok on the classpath and run:

```
spotbugs -textui -effort:max -low <classes>
```

**Expected behavior**
Two `DMI_RANDOM_USED_ONLY_ONCE` findings are reported against `numberProvider()` and `boundedProvider(int)` in the `AFTER` variant, matching the `BEFORE` variant (bytecode is equivalent).

**Actual behavior**
SpotBugs reports zero findings for the `AFTER` variant, even though the `new Random().nextInt(...)` call sites are unchanged.

**Environment**

- Java version: JDK 21 (Eclipse Temurin)
- SpotBugs Version: 4.9.8
- How do you use SpotBugs: CLI

**Comment by Frogzinkkj** [comment]
Hi, I saw this issue and I really liked it. I would love to help, could you assign me the task?

**Comment by JuditKnoll** [comment]
> Hi, I saw this issue and I really liked it. I would love to help, could you assign me the task?

Feel free to work on this issue without it assigned it to you. If you have a solution open a PR referencing this issue.
